front50-core/src/main/groovy/com/netflix/spinnaker/front50/model/StorageServiceSupport.java

```diff
@@ -34,58 +33,49 @@
 import java.util.stream.Collectors;
 
 
-public abstract class BucketDAO<T extends Timestamped> {
+public abstract class StorageServiceSupport<T extends Timestamped> {
   public static long HEALTH_MILLIS = 45000;
   private final Logger log = LoggerFactory.getLogger(getClass());
   protected final AtomicReference<Set<T>> allItemsCache = new AtomicReference<>();
 
+  private final ObjectType objectType;
   private final StorageService service;
-  private final String daoTypeName;
-  private final String rootFolder;
-  private final Class<T> serializedClass;
   private final Scheduler scheduler;
   private final int refreshIntervalMs;
 
   private long lastRefreshedTime;
 
-  public BucketDAO(Class<T> serializedClass, String daoTypeName, String basePath,
-                   StorageService service,
-                   Scheduler scheduler, int refreshIntervalMs) {
-    this.serializedClass = serializedClass;
-    this.daoTypeName = daoTypeName;
+  public StorageServiceSupport(ObjectType objectType,
+                               StorageService service,
+                               Scheduler scheduler,
+                               int refreshIntervalMs) {
+    this.objectType = objectType;
     this.service = service;
-    this.rootFolder = basePath + '/' + daoTypeName + '/';
     this.scheduler = scheduler;
     this.refreshIntervalMs = refreshIntervalMs;
   }
 
   @PostConstruct
   void startRefresh() {
-    // TODO(ewiseblatt): 20160526
-    // Make this start executing now but in another thread.
-    // I dont know how to say that using this API so am
-    // using a timer instead.
-    Observable
-        .timer(0, TimeUnit.MILLISECONDS, scheduler)
-        .subscribe(interval -> {
-          try {
-            log.info("Warming Cache");
-            refresh();
-          } catch (Exception e) {
-            log.error("Unable to refresh: {}", e);
-          }
-        });
-    Observable
+    if (refreshIntervalMs > 0) {
+      try {
+        log.info("Warming Cache");
+        refresh();
```

Comment:
I did swap this such that it runs on the main thread ... I don't believe it'll materially increase boot time.